rfc/2023-12-custom-operator-extension.md

```diff
@@ -0,0 +1,224 @@
+# Custom operator extension
+
+**Status:** { **RFC** | ~~Final | POR | Inactive~~ }
+
+**Author:** Denis Vieriu
+
+**Last Update:** 2023-12-05
+
+# Summary
+[summary]: #summary
+
+Today, custom kernels have no knowledge of the device they are running, and it makes very hard for ExecuTorch delegates to detect when a custom operator is being used. This document proposes two solutions on how to extend the custom kernels, so that a delegate can intercept when a custom op is running, and if it can improve it.
+
+# Motivation
+[motivation]: #motivation
+
+Custom kernels implemented by the users currently have no knowledge about the device they are running on and when they are being dispatch. This makes it very hard to share resources between a delegate and a custom operation. For example, consider 3 lowered modules, running in following order:
```

Comment:
I am gonna type some comments and summarize it later. Op execution in executorch is device agnostic. There is not concept of device specific tensors. A custom op can run on any device but the input and output tensors must be on the memory that is accessible by dereferencing the pointer. While one can argue whether we want to introduce device specific tensors, current status is that they are not device specific.